snapper/BtrfsUtils.cc

```diff
@@ -249,18 +251,92 @@ namespace snapper
 	    return args.treeid;
 	}
 
+
+	bool
+	does_subvolume_exist(int fd, subvolid_t subvolid)
+	{
+	    struct btrfs_ioctl_search_args args;
+	    struct btrfs_ioctl_search_key* sk = &args.key;
+
+	    sk->tree_id = BTRFS_ROOT_TREE_OBJECTID;
+	    sk->min_objectid = subvolid;
+	    sk->max_objectid = subvolid;
+	    sk->min_type = BTRFS_ROOT_ITEM_KEY;
+	    sk->max_type = BTRFS_ROOT_ITEM_KEY;
+	    sk->min_offset = 0;
+	    sk->max_offset = (u64) -1;
+	    sk->min_transid = 0;
+	    sk->max_transid = (u64) -1;
+	    sk->nr_items = 1;
+
+	    if (ioctl(fd, BTRFS_IOC_TREE_SEARCH, &args) != 0)
+		throw runtime_error_with_errno("ioctl(BTRFS_IOC_TREE_SEARCH) failed", errno);
```

Comment:
Since this is used so often: maybe use a function or a macro and thus repeat duplication?
Maybe even one that does the ioctl, only receiving fd, the ioctl name, and args?